iothub/device/src/DeviceClient.ConventionBasedOperations.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    /// Contains methods that a convention based device can use to send telemetry to the service,
+    /// respond to commands and perform operations on its properties.
+    /// </summary>
+    /// <threadsafety static="true" instance="true" />
+    public partial class DeviceClient : IDisposable
```

Comment:
I updated `DeviceClient` to be a partial class, so that we could separate out the convention-based operations from the non-convention based ones. I preferred to use this over using regions on the existing DeviceClient class.
Let me know if there are any concerns.